### ISSUE TITLE
Yet another broadcasting error

### ISSUE THREAD

**Issue by dfdx** [JuliaGPU/CuArrays.jl#42]
Earlier I posted a [similar issue](https://github.com/FluxML/NNlib.jl/issues/20) to FluxML/NNlib.jl, but this one is unrelated to that repo. 

The following code:
```
using CuArrays

A = cu(rand(Float32, 10, 100))
B = cu(ones(Float32, 10, 100))
C = cu(ones(Float32, 1, 100)) 
D = cu(ones(Float32, 1, 100)) 
E = cu(rand(Float32, 10, 100))

CUDAnative.exp.(1 ./ A) .* (B .* (C ./ D)) .+ E
```
generates this error: 

```
WARNING: Method definition convert(Type{LLVM.LLVMType}, Type{T} where T) in module Interop at /home/dfdx/.julia/v0.6/LLVM/src/interop/base.jl:54 overwritten in module CUDAnative at /home/dfdx/.julia/v0.6/CUDAnative/src/cgutils.jl:159.


warning: ignoring debug info with an invalid version (0) in #1
warning: ignoring debug info with an invalid version (0) in 
ERROR: LoadError: LLVM error: All DICompileUnits must be listed in llvm.dbg.cu

Stacktrace:
 [1] verify(::LLVM.Module) at /home/dfdx/.julia/v0.6/LLVM/src/analysis.jl:11
 [2] #add_entry!#26(::Bool, ::Function, ::LLVM.Module, ::Any, ::Any) at /home/dfdx/.julia/v0.6/CUDAnative/src/jit.jl:251
 [3] (::CUDAnative.#kw##add_entry!)(::Array{Any,1}, ::CUDAnative.#add_entry!, ::LLVM.Module, ::Any, ::Any) at ./<missing>:0
 [4] #compile_function#51(::Bool, ::Function, ::Any, ::Any, ::VersionNumber) at /home/dfdx/.julia/v0.6/CUDAnative/src/jit.jl:402
 [5] cufunction(::CUDAdrv.CuDevice, ::Any, ::Any) at /home/dfdx/.julia/v0.6/CUDAnative/src/jit.jl:465
 [6] macro expansion at /home/dfdx/.julia/v0.6/CUDAnative/src/execution.jl:108 [inlined]
 [7] _cuda(::Tuple{Int64,Int64}, ::Int64, ::CUDAdrv.CuStream, ::CuArrays.#broadcast_kernel, ::##1#2, ::CUDAnative.CuDeviceArray{Float32,2,CUDAnative.AS.Global}, ::NTuple{5,Tuple{Bool,Bool}}, ::NTuple{5,Tuple{Int64,Int64}}, ::CUDAnative.CuDeviceArray{Float32,2,CUDAnative.AS.Global}, ::NTuple{4,CUDAnative.CuDeviceArray{Float32,2,CUDAnative.AS.Global}}) at /home/dfdx/.julia/v0.6/CUDAnative/src/execution.jl:80
 [8] _broadcast! at /home/dfdx/.julia/v0.6/CuArrays/src/broadcast.jl:22 [inlined]
 [9] broadcast_t(::Function, ::Type{T} where T, ::Tuple{Base.OneTo{Int64},Base.OneTo{Int64}}, ::CuArray{Float32,2}, ::CuArray{Float32,2}, ::CuArray{Float32,2}, ::CuArray{Float32,2}, ::CuArray{Float32,2}) at /home/dfdx/.julia/v0.6/CuArrays/src/broadcast.jl:37
 [10] broadcast_c(::Function, ::Type{CuArrays.CuArray}, ::CuArray{Float32,2}, ::CuArray{Float32,2}, ::Vararg{CuArray{Float32,2},N} where N) at /home/dfdx/.julia/v0.6/CuArrays/src/broadcast.jl:58
 [11] broadcast(::Function, ::CuArray{Float32,2}, ::CuArray{Float32,2}, ::CuArray{Float32,2}, ::Vararg{CuArray{Float32,2},N} where N) at ./broadcast.jl:455
 [12] include_from_node1(::String) at ./loading.jl:576
 [13] include(::String) at ./sysimg.jl:14
while loading /home/dfdx/Downloads/broadcast_fail.jl, in expression starting on line 10
```
I use Julia 0.6.2, CUDAnative 0.5.3 and latest master of CuArrays. I'm open to test it on Julia 0.7 and latest CUDAnative master if it's assumed to be fixed there, but right now this setup seems to be broken (or at least I couldn't build latest CUDAnative on my machine), so it will take time to resolve,

**Comment by maleadt** [comment]
This is a problem with CUDAnative, and should be fixed in v0.5.4. The example works for me.
Also, feel free to tag me when a problem points to CUDAnative; I'm not watching all JuliaGPU repo's 🙂

**Comment by dfdx** [comment]
Yeah, I had doubts about where to post it :) Thanks a lot for the fix!
